### PR TITLE
Add Dockerfile, Makefile targets and install_deps.sh

### PR DESCRIPTION
**Pull request by krsna1729**
- Move to Ubuntu 18.04
- Added simple script to install all dependencies
  - Use packages for hyperscan
- Added a multi-stage Dockerfile to build SPGW-C/U
- Added docker related targets to top-level Makefile

Fixes: #6, #7
Based-on: #31

Signed-off-by: Saikrishna Edupuganti <saikrishna.edupuganti@intel.com>

### CODE REVIEW

**Comment by krsna1729** [comment]
Broke down the earlier patch so this change does not affect current workflow and source files.

**Comment by krsna1729** [comment]
This dockerfile will be useable when following issues/PRs resolve

#3, #18, #19, #20

File: install_deps.sh [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+(return 2>/dev/null) && echo "Sourced" && return
+
+set -o errexit
+set -o pipefail
+set -o nounset
+
+install_pkg_deps
+install_dpdk
+echo "Dependency install complete"
```
Comment (krsna1729) [inline install_deps.sh]: TODO:
- [ ] Download [SGX](https://github.com/intel/linux-sgx/tags)

**Comment by omecproject** [comment]
Can one of the admins verify this patch?

**Comment by omecproject** [comment]
Can one of the admins verify this patch?

**Comment by omecproject** [comment]
Can one of the admins verify this patch?

**Comment by krsna1729** [comment]
ok to test

**Comment by krsna1729** [comment]
retest this please

**Pull request merged**